src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -70,11 +72,16 @@ def execute(self):
     with self.invalidated(binaries, invalidate_dependents=True) as invalidation_check:
       python_deployable_archive = self.context.products.get('deployable_archives')
       python_pex_product = self.context.products.get('pex_archives')
+      precedence = python_setup.default_precedence[:]
       for vt in invalidation_check.all_vts:
         pex_path = os.path.join(vt.results_dir, '{}.pex'.format(vt.target.name))
         if not vt.valid:
           self.context.log.debug('cache for {} is invalid, rebuilding'.format(vt.target))
-          self._create_binary(vt.target, vt.results_dir)
+          if vt.target.build is True:
+            precedence = python_setup._allow_builds(precedence)
+          elif vt.target.build is False:
+            precedence = python_setup._no_allow_builds(precedence)
```

Comment:
these calls are currently violating the "private method" convention - so probably need to drop the `_` prefix at the Subsystem method level.